median/aggregated_attribute_observation.go

```diff
@@ -0,0 +1,46 @@
+package median
+
+import (
+	"math/big"
+	"sort"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+)
+
+type aggregatedAttributedObservation struct {
+	Timestamp       uint32
+	Observers       [32]commontypes.OracleID
+	Observations    []*big.Int
+	JuelsPerFeeCoin *big.Int
+}
+
+func aggregate(observations []median.ParsedAttributedObservation) *aggregatedAttributedObservation {
+	// defensive copy
+	n := len(observations)
+	observations = append([]median.ParsedAttributedObservation{}, observations...)
+
+	aggregated := &aggregatedAttributedObservation{Observations: make([]*big.Int, len(observations))}
+
+	sort.Slice(observations, func(i, j int) bool {
+		return observations[i].Timestamp < observations[j].Timestamp
+	})
```

Comment:
You could avoid the reflection in `sort.Slice` by using `slices.SortFunc`:
```suggestion
	slices.SortFunc(observations, func(a, b ParsedAttributedObservation) int {
		return cmp.Compare(a[i].Timestamp, b[j].Timestamp)
	})
```